jupyterlab_git/git.py

```diff
@@ -955,9 +955,23 @@ async def show_prefix(self, path):
             cmd,
             cwd=path,
         )
+
         if code == 0:
-            result = {"code": code, "path": my_output.strip("\n")}
-            return result
+            relative_git_path = my_output.strip("\n")
+            relative_server_path = os.path.relpath(path, contents_manager.root_dir)
+
+            result = {
+                "code": code,
+                "path": relative_git_path,
+                "relative_path": relative_server_path,
+            }
+
+            if relative_server_path == "." and relative_git_path == "":
+                return result
+            elif relative_git_path.split("/")[:-1] == relative_server_path.split("\\"):
+                return result
+            else:
+                return {"code": code, "path": None}
```

Comment:
Ok I was playing around a bit also to try  to avoid the `split("\\")` that I think will only work for windows.

Here is an alternative suggestion

```suggestion
            relative_git_path = my_output.strip("\n")
            repository_path = path[:-len(relative_git_path)]

            try:
                # Raise an error is the repository_path is not a subpath of root_dir
                Path(repository_path).absolute().relative_to(Path(contents_manager.root_dir).absolute())
            except ValueError:
                return {"code": code, "path": None}
            else:
                result = {
                    "code": code,
                    "path": relative_git_path,
                }
```

What do you think?